packages/cadl-ranch-specs/http/encode/datetime/main.tsp

```diff
@@ -0,0 +1,282 @@
+import "@typespec/http";
+import "@azure-tools/cadl-ranch-expect";
+import "@azure-tools/typespec-client-generator-core";
+
+using TypeSpec.Http;
+using Azure.ClientGenerator.Core;
+
+@doc("Test for encode decorator on datetime.")
+@supportedBy("dpg")
+@scenarioService("/encode/datetime")
+namespace Encode.Datetime;
+
+@encode(DateTimeKnownEncoding.rfc7231)
+scalar rfc7231Datetime extends utcDateTime;
+
+@encode(DateTimeKnownEncoding.unixTimestamp, int64)
+scalar unixTimestampDatetime extends utcDateTime;
+
+@operationGroup
+@route("/query")
+namespace Query {
+  @route("/default")
+  @scenario
+  @scenarioDoc("""
+  Test default encode for datetime query parameter.
+  Expected query parameter: 
+  value=2022-08-26T18:38:00.000Z
+  """)
+  op default(
+    @query
+    value: utcDateTime
+  ): NoContentResponse;
+
+  @route("/rfc3339")
+  @scenario
+  @scenarioDoc("""
+  Test rfc3339 encode for datetime query parameter.
+  Expected query parameter: 
+  value=2022-08-26T18:38:00.000Z
+  """)
+  op rfc3339(
+    @query
+    @encode(DateTimeKnownEncoding.rfc3339)
+    value: utcDateTime
+  ): NoContentResponse;
+
+  @route("/rfc7231")
+  @scenario
+  @scenarioDoc("""
+  Test rfc7231 encode for datetime query parameter.
+  Expected query parameter: 
+  value=Fri, 26 Aug 2022 14:38:00 GMT
+  """)
+  op rfc7231(
+    @query
+    @encode(DateTimeKnownEncoding.rfc7231)
+    value: utcDateTime
+  ): NoContentResponse;
+
+  @route("/unix-timestamp")
+  @scenario
+  @scenarioDoc("""
+  Test unixTimestamp encode for datetime query parameter.
+  Expected query parameter: 
+  value=1686566864
+  """)
+  op unixTimestamp(
+    @query
+    @encode(DateTimeKnownEncoding.unixTimestamp, int64)
+    value: utcDateTime
+  ): NoContentResponse;
+
+  @route("/rfc7231-array")
+  @scenario
+  @scenarioDoc("""
+  Test rfc7231 encode for datetime array query parameter.
+  Expected query parameter: 
+  value=Fri, 26 Aug 2022 14:38:00 GMT, Fri, 26 Aug 2022 16:38:00 GMT
```

Comment:
Same question as csv header, is it a meaningful case?

Do we had to use `format=csv` when data contains comma? Can it be multi or piple?

Reply:
No. Header could only use csv.

Reply:
This is query.

Reply:
For Python, we do url encoding for each of the array item, then contact them with raw `,`. It means the final url will be `k=<url encoded v1>,<url encoded v2>` and it could be parsed by cadl-ranch server correctly. I don't know if this is a common pattern.

Reply:
Change to unixTimestamp to prevent possible encode issue.